### PR TITLE
New: seqan3/3.1.0

### PR DESCRIPTION
**Pull request by eseiler**
Library name and version:  **seqan3/3.1.0**

[seqan3](https://github.com/seqan/seqan3) is a header-only C++ library for sequence analysis

I am adding this recipe since a user requested it to be added: https://github.com/seqan/seqan3/issues/2974

It is very likely that we will add applications in the future that rely on seqan3 as a dependency, we have by now done so for other package repositories.

There is already a `seqan` recipe, it is for the old 2.x.x versions. _SeqAn3 is a complete rewrite_.

I am uncertain whether it should be added as a new version for the `seqan` recipe, or if it may exist as a separate package. 
**I am happy to handle this in whatever way you deem appropriate.**

Currently, we have some standalone packages (alongside the "old" `seqan` package):

* Conda: https://anaconda.org/bioconda/seqan3
* Debian: https://packages.debian.org/sid/libseqan3-dev
* MacPorts: https://ports.macports.org/port/seqan3/

And only one where both seqan 2.x.x and 3.x.x are one package (with `seqan@2` and `seqan@3`):
* Brew: https://github.com/brewsci/homebrew-bio/blob/develop/Formula/seqan%403.rb
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by AndreyMlashkin — APPROVED** [review]

File: recipes/seqan3/all/test_package/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import os
+from conans import ConanFile, CMake, tools
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake"
```
Comment (madebr) [inline recipes/seqan3/all/test_package/conanfile.py:7]: Please add the `cmake_find_package` generator +
 use `seqan3::seqan3` in the test package CMake script.

https://github.com/seqan/seqan3/blob/24a24419d2e8e7b936c8962c63c5c514017059ed/build_system/seqan3-config.cmake#L498-L511

**Review by danimtb — APPROVED** [review]

File: recipes/seqan3/all/test_package/conanfile.py (hunk 1 above)
Comment (SpaceIm) [reply recipes/seqan3/all/test_package/conanfile.py:7]: https://github.com/conan-io/conan-center-index/pull/10924

**Comment by SpaceIm** [comment]
What about optional dependencies like bzip2, zlib, cereal and lemon? https://github.com/seqan/seqan3/blob/master/build_system/seqan3-config.cmake#L28-L33

**Comment by eseiler** [comment]
Thanks for the follow-up PR, @SpaceIm !

> What about mandatory (sdsl-lite & range-v3) & optional dependencies (bzip2, zlib, cereal and lemon)? 

The library dependencies are a bit finicky with 3.1.0:
* sdsl-lite should be fine since we ship 3.0.0 in the package containing submodules
* range-v3 is usually used on a specific commit (i.e., not a tag) as we require certain fixes
* cereal 1.3.1 should work
* bzip2, zlib should be added
* lemon we do not actually use yet. It's more of a legacy thing because we planned to use it in the beginning (and might in the future). So, including it does nothing.

We are about to release 3.2 soonish:
* We'll do a new release of sdsl-lite with gcc12 support
* We drop range-v3
* I'll have to check cereal 1.3.1, I think we are ahead of 1.3.1 because of a fix for gcc12 - this should be fixable by a patch?

Once we release, I'll upgrade the package and try to use the provided libs instead of using our catch-all package.

**Pull request merged**